File: batallanaval.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

const input = (pregunta) => {
    return new Promise((resolve) => {
      rl.question(pregunta, (respuesta) => resolve(respuesta));
    });
  };
  

function crearTablero(a, b) {
  const filas = {};
  for (let j = 0; j < b; j++) {
    filas[j] = Array(a).fill(".");
  }
  return filas;
}

function mostrarTablero(tablero) {
  for (const fila of Object.values(tablero)) {
    console.log(fila.join(""));
  }
  console.log("");
}

const tablero = crearTablero(10, 10);
mostrarTablero(tablero);


const main = async () => {
  const numBarcos = parseInt(await input("Cuanto barcos desea colocar en el tablero?: "));
  console.log("");

  for (let i = 0; i < numBarcos; i++) {
    console.log(`BARCO NUMERO ${i + 1}`);
    const tamaño = parseInt(await input("Ingrese tamaño del barco: "));
    const orientacion = await input("Ingrese horientacion del barco: ");
    const fila = parseInt(await input("Ingrese fila donde posicionar el barco: "));
    const columna = parseInt(await input("Ingrese columna donde posicionar el barco: "));

    if (orientacion === "horizontal") {
      for (let j = 0; j < tamaño; j++) {
        tablero[fila][columna + j] = "X";
      }
      mostrarTablero(tablero);
    } else if (orientacion === "vertical") {
      for (let j = 0; j < tamaño; j++) {
        tablero[fila + j][columna] = "X";
      }
      mostrarTablero(tablero);
    }
  }

  rl.close(); 
};

main();

File: batallanaval.py
def crear_tablero(a, b):
    filas = {}
    for j in range(0, b):
        filas[j] = ["."] * a
    return filas

def mostrar_tablero(tablero):
    for i in range(len(tablero)):
        print(tablero[i])
    print("")

tablero = crear_tablero(10, 10)
mostrar_tablero(tablero)

num_barcos = int(input("Cuanto barcos desea colocar en el tablero?: "))
print("")
for i in range(0,num_barcos):
    print(f"BARCO NUMERO {i+1}")
    tamaño = int(input("Ingrese tamaño del barco: "))
    orientacion = input("Ingrese horientacion del barco: ")
    fila = int(input("Ingrese fila donde posicionar el barco: "))
    columna = int(input("Ingrese columna donde posicionar el barco"))

    if orientacion == "horizontal":
        for i in range(0,tamaño):
            tablero[fila][columna+i] = "X"
        mostrar_tablero(tablero)

    if orientacion == "vertical":
        for i in range(0,tamaño):
            tablero[fila+i][columna] = "X"
        mostrar_tablero(tablero)